medusa/server/web/core/base.py

```diff
@@ -226,10 +226,13 @@ def redirect(self, url, permanent=False, status=None):
                                             utf8(url)))
 
     def get_current_user(self):
-        user_cookie = self.get_secure_cookie(app.SECURE_TOKEN)
-        if user_cookie:
-            return json.dumps(user_cookie)
-        return None
+        if not isinstance(self, UI):
+            if app.WEB_USERNAME and app.WEB_PASSWORD:
+                return self.get_secure_cookie('medusa_user')
+            else:
+                return True
+        else:
+            return None
```

Comment:
Why have you changed it all back? Why are we checking if it's an instance of UI? `app.SECURE_TOKEN` should be used instead of `medusa_user`.